aws/stack/register_middleware.go

```diff
@@ -0,0 +1,54 @@
+package stack
+
+import (
+	"github.com/aws/aws-sdk-go-v2/aws/retry"
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	"github.com/awslabs/smithy-go/middleware"
+	smithyhttp "github.com/awslabs/smithy-go/transport/http"
+)
+
+// AwsRetryMiddlewareConfig interface for retry middleware config
+type AwsRetryMiddlewareConfig interface {
+	GetRetryer() retry.Retryer
+}
+
+// AddRetryMiddlewares adds retry middleware to operation middleware stack
+func AddRetryMiddlewares(stack *middleware.Stack, cfg AwsRetryMiddlewareConfig,
+	optsFn ...func(*retry.AttemptMiddleware, *retry.MetricsHeaderMiddleware)) {
+	attemptMiddleware := retry.NewAttemptMiddleware(cfg.GetRetryer(), smithyhttp.RequestCloner)
+	metricsHeaderMiddleware := retry.MetricsHeaderMiddleware{}
+	for _, fn := range optsFn {
+		fn(&attemptMiddleware, &metricsHeaderMiddleware)
+	}
+
+	stack.Finalize.Add(attemptMiddleware, middleware.After)
+	stack.Finalize.Add(metricsHeaderMiddleware, middleware.After)
+}
+
+// AddUnsignedPayloadMiddlewares adds UnsignedPayloadMiddleware to the operation middleware stack
+func AddUnsignedPayloadMiddlewares(stack *middleware.Stack, optFns ...func(*v4.UnsignedPayloadMiddleware)) {
+	unsignedPayloadMiddleware := v4.UnsignedPayloadMiddleware{}
+	for _, fn := range optFns {
+		fn(&unsignedPayloadMiddleware)
+	}
+
+	stack.Finalize.Add(&unsignedPayloadMiddleware, middleware.After)
+}
```

Comment:
Probably won't want functional args to be the default when they don't  have anything to configure.